src/Linux/Host/Broker/Impl/Services/LinuxRHostProcessService.cs

```diff
@@ -45,78 +45,32 @@ class LinuxRHostProcessService : IRHostProcessService {
-                { "R_BZIPCMD"               , GetCurrentOrDefault("R_BZIPCMD")},
-                { "R_DOC_DIR"               , interpreter.RInterpreterInfo.DocPath},
-                { "R_GZIPCMD"               , GetCurrentOrDefault("R_GZIPCMD")},
                 { "R_HOME"                  , interpreter.RInterpreterInfo.InstallPath},
-                { "R_INCLUDE_DIR"           , interpreter.RInterpreterInfo.IncludePath},
-                { "R_LIBS_SITE"             , siteLibrary},
-                { "R_PAPERSIZE"             , GetCurrentOrDefault("R_PAPERSIZE")},
-                { "R_PAPERSIZE_USER"        , GetCurrentOrDefault("R_PAPERSIZE_USER")},
-                { "R_PDFVIEWER"             , GetCurrentOrDefault("R_PDFVIEWER")},
-                { "R_PRINTCMD"              , GetCurrentOrDefault("R_PRINTCMD")},
-                { "R_RD4PDF"                , GetCurrentOrDefault("R_RD4PDF")},
-                { "R_SHARE_DIR"             , interpreter.RInterpreterInfo.RShareDir},
-                { "R_TEXI2DVICMD"           , GetCurrentOrDefault("R_TEXI2DVICMD")},
-                { "R_UNZIPCMD"              , GetCurrentOrDefault("R_UNZIPCMD")},
-                { "R_ZIPCMD"                , GetCurrentOrDefault("R_ZIPCMD")},
-                { "SED"                     , GetCurrentOrDefault("SED")},
-                { "SHELL"                   , GetCurrentOrDefault("SHELL")},
-                { "SHLVL"                   , GetCurrentOrDefault("SHLVL")},
-                { "TAR"                     , GetCurrentOrDefault("TAR")},
                 { "USER"                    , Utility.GetUnixUserName(userName)},
             };
+
+            // set optional environment variables if available
+            string[] optionalVariables = { "LD_LIBRARY_PATH", "LN_S", "R_ARCH", "R_BROWSER", "R_BZIPCMD", "R_GZIPCMD", "R_LIBS_SITE", "R_INCLUDE_DIR", "R_DOC_DIR", "R_PAPERSIZE", "R_PAPERSIZE_USER", "R_PDFVIEWER", "R_PRINTCMD", "R_RD4PDF", "R_SHARE_DIR", "R_TEXI2DVICMD", "R_UNZIPCMD", "R_ZIPCMD", "SED", "SHELL", "SHLVL", "TAR" };
+            foreach(string key in optionalVariables) {
+                SetEnvironmentVaraibleIfAvailable(environment, key);
+            }
+
+            // TODO: provide RTVS_LD_LIBRARY_PATH
             return environment;
         }
 
-        private static string GetCurrentOrDefault(string key) {
+        private void SetEnvironmentVaraibleIfAvailable(Dictionary<string,string> environment, string key) {
             var value = Environment.GetEnvironmentVariable(key);
-            if (string.IsNullOrEmpty(value) && !_defaultEnvironment.TryGetValue(key, out value)) {
-                return string.Empty;
+            if (!string.IsNullOrEmpty(value)) {
+                environment.Add(key, value);
             }
-            return value;
         }
-        private static readonly IDictionary<string, string> _defaultEnvironment = new Dictionary<string,string>(){
```

Comment:
We shouldn't be in the business of providing any of these - if they're not inherited from the parent environment, R itself chooses the appropriate defaults (which may well be different).